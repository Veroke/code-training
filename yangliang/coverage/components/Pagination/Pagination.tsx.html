
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for components/Pagination/Pagination.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> / <a href="index.html">components/Pagination</a> Pagination.tsx</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">67.47% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>83/123</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">60% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>6/10</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">20% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>1/5</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">67.47% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>83/123</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a>
<a name='L94'></a><a href='#L94'>94</a>
<a name='L95'></a><a href='#L95'>95</a>
<a name='L96'></a><a href='#L96'>96</a>
<a name='L97'></a><a href='#L97'>97</a>
<a name='L98'></a><a href='#L98'>98</a>
<a name='L99'></a><a href='#L99'>99</a>
<a name='L100'></a><a href='#L100'>100</a>
<a name='L101'></a><a href='#L101'>101</a>
<a name='L102'></a><a href='#L102'>102</a>
<a name='L103'></a><a href='#L103'>103</a>
<a name='L104'></a><a href='#L104'>104</a>
<a name='L105'></a><a href='#L105'>105</a>
<a name='L106'></a><a href='#L106'>106</a>
<a name='L107'></a><a href='#L107'>107</a>
<a name='L108'></a><a href='#L108'>108</a>
<a name='L109'></a><a href='#L109'>109</a>
<a name='L110'></a><a href='#L110'>110</a>
<a name='L111'></a><a href='#L111'>111</a>
<a name='L112'></a><a href='#L112'>112</a>
<a name='L113'></a><a href='#L113'>113</a>
<a name='L114'></a><a href='#L114'>114</a>
<a name='L115'></a><a href='#L115'>115</a>
<a name='L116'></a><a href='#L116'>116</a>
<a name='L117'></a><a href='#L117'>117</a>
<a name='L118'></a><a href='#L118'>118</a>
<a name='L119'></a><a href='#L119'>119</a>
<a name='L120'></a><a href='#L120'>120</a>
<a name='L121'></a><a href='#L121'>121</a>
<a name='L122'></a><a href='#L122'>122</a>
<a name='L123'></a><a href='#L123'>123</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span></td><td class="text"><pre class="prettyprint lang-js">import { defineComponent, ref, computed, onMounted, onErrorCaptured } from "vue";
import { addLog } from "../../log/log";
import Pager from './pager/Pager';
import './pagination.css';
&nbsp;
export type PaginationProps = {
  current: number;
  pageSize: number;
  total: number;
  showSizeChange: () =&gt; {};
  showQuickJumper: boolean;
  onChange: (page:number) =&gt; {};
}
&nbsp;
export default defineComponent({
    name: "Pagination",
    props: [
      'current',
      'pageSize',
      'total',
      'showSizeChange',
      'showQuickJumper',
      'onChange',
    ],
    setup(props: PaginationProps, { attrs, emit, slots }) {
      onMounted(()=&gt;{
        addLog({
          level: "info",
          message: {
            component: "Pagination",
            props,
            msg: 'Some One Use Pagination Component'
          }
        })
      })
&nbsp;
      onErrorCaptured((err) =&gt; {
<span class="cstat-no" title="statement not covered" >        addLog({</span>
<span class="cstat-no" title="statement not covered" >          level: "error",</span>
<span class="cstat-no" title="statement not covered" >          message: {</span>
<span class="cstat-no" title="statement not covered" >            component: "Pagination",</span>
<span class="cstat-no" title="statement not covered" >            props,</span>
<span class="cstat-no" title="statement not covered" >            msg: err</span>
<span class="cstat-no" title="statement not covered" >          }</span>
<span class="cstat-no" title="statement not covered" >        })</span>
<span class="cstat-no" title="statement not covered" >      })</span>
      const currentPage = ref(props.current);
      const pageCount = ref(Math.ceil(props.total / props.pageSize));
      const pagesList = computed(() =&gt; {
      const tempPageList = [];
        for (let i = 0; i &lt; pageCount.value; i++) <span class="branch-0 cbranch-no" title="branch not covered" >{</span>
<span class="cstat-no" title="statement not covered" >          tempPageList.push(&lt;Pager key={i + 1} </span>
<span class="cstat-no" title="statement not covered" >                                   page={i + 1}</span>
<span class="cstat-no" title="statement not covered" >                                   onClick={(<span class="fstat-no" title="function not covered" >page:number) =&gt; {</span></span>
<span class="cstat-no" title="statement not covered" >                                      emit('change', page);</span>
<span class="cstat-no" title="statement not covered" >                                      currentPage.value = page;</span>
<span class="cstat-no" title="statement not covered" >                                   }}</span>
<span class="cstat-no" title="statement not covered" >                                   active={Number(currentPage.value) === (i + 1)} /&gt;);</span>
<span class="cstat-no" title="statement not covered" >        }</span>
<span class="cstat-no" title="statement not covered" >        return tempPageList</span>
<span class="cstat-no" title="statement not covered" >      });</span>
      const preClick = <span class="fstat-no" title="function not covered" >() =&gt; {</span>
<span class="cstat-no" title="statement not covered" >        if(currentPage.value &gt; 1){</span>
<span class="cstat-no" title="statement not covered" >          currentPage.value -= 1;</span>
<span class="cstat-no" title="statement not covered" >          emit('change', currentPage.value);</span>
<span class="cstat-no" title="statement not covered" >        }</span>
<span class="cstat-no" title="statement not covered" >      }</span>
      const nextClick = <span class="fstat-no" title="function not covered" >() =&gt; {</span>
<span class="cstat-no" title="statement not covered" >        if(currentPage.value &lt; pageCount.value){</span>
<span class="cstat-no" title="statement not covered" >          currentPage.value += 1;</span>
<span class="cstat-no" title="statement not covered" >          emit('change', currentPage.value);</span>
<span class="cstat-no" title="statement not covered" >        }</span>
<span class="cstat-no" title="statement not covered" >      }</span>
      const preDisabled = computed(() =&gt; {
        return <span class="branch-0 cbranch-no" title="branch not covered" >currentPage.value === 1 ? 'disabled' : '';</span>
      })
      const nextDisabled = computed(() =&gt; {
        return <span class="branch-0 cbranch-no" title="branch not covered" >currentPage.value === pageCount.value ? 'disabled' : '</span>';
      })
      let timer: any = null;
      
      const showSizeChange = (<span class="fstat-no" title="function not covered" >e: Event) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >        timer = setTimeout(() =&gt; {</span>
<span class="cstat-no" title="statement not covered" >          if(typeof timer === 'number'){</span>
<span class="cstat-no" title="statement not covered" >            clearTimeout(timer)</span>
<span class="cstat-no" title="statement not covered" >            const page = (e.target as any)?.value;</span>
<span class="cstat-no" title="statement not covered" >            if(typeof page !== 'number' &amp;&amp; page &gt;= 1 &amp;&amp; page &lt;= pageCount.value) {</span>
<span class="cstat-no" title="statement not covered" >              currentPage.value = page;</span>
<span class="cstat-no" title="statement not covered" >              emit('change', currentPage.value);</span>
<span class="cstat-no" title="statement not covered" >            }</span>
<span class="cstat-no" title="statement not covered" >          }</span>
<span class="cstat-no" title="statement not covered" >        },300);</span>
<span class="cstat-no" title="statement not covered" >      }</span>
        
      return () =&gt; {
        return (
          &lt;ul&gt;
            &lt;li class={`pager ${preDisabled.value}`} 
                onClick={preClick}&gt;
              &lt;svg focusable="false" class="pre" data-icon="left" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"&gt;&lt;path d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"&gt;
                &lt;/path&gt;
              &lt;/svg&gt;
            &lt;/li&gt;
            {
              pagesList.value.map(pager =&gt; pager)
            }
            &lt;li class={`pager ${nextDisabled.value}`}
                onClick={nextClick}&gt;
                &lt;svg focusable="false" class="next" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"&gt;&lt;path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"&gt;
                  &lt;/path&gt;
                &lt;/svg&gt;
            &lt;/li&gt;
            {
<span class="branch-0 cbranch-no" title="branch not covered" >              props.</span>showQuickJumper &amp;&amp; 
              &lt;div class="show-quick-jumper"&gt;
                跳至&lt;input type="text" onInput={showSizeChange} /&gt;页
              &lt;/div&gt;
            }
          &lt;/ul&gt;
        );
      };
    },
});</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Sat May 28 2022 17:53:17 GMT+0800 (中国标准时间)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    