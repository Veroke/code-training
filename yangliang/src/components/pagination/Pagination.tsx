import { defineComponent, ref, computed, onMounted, onErrorCaptured } from "vue";
import { addLog } from "../../log/log";
import { type PaginationProps, paginationProps } from "../types";
import Pager from './pager/Pager';
import './pagination.css';

export default defineComponent({
    name: "Pagination",
    props: paginationProps,
    setup(props: PaginationProps, { attrs, emit, slots }) {
      onMounted(()=>{
        addLog({
          level: "info",
          message: {
            component: "Pagination",
            props,
            msg: 'Some One Use Pagination Component'
          }
        })
      })

      onErrorCaptured((err) => {
        addLog({
          level: "error",
          message: {
            component: "Pagination",
            props,
            msg: err
          }
        })
      })
      const currentPage = ref(props.current);
      const pageCount = ref(Math.ceil(props.total / props.pageSize));
      const pagesList = computed(() => {
      const tempPageList = [];
        for (let i = 0; i < pageCount.value; i++) {
          tempPageList.push(<Pager key={i + 1} 
                                   page={i + 1}
                                   onClick={(page:number) => {
                                      emit('change', page);
                                      currentPage.value = page;
                                   }}
                                   active={currentPage.value === i + 1} />);
        }
        return tempPageList
      });
      const preClick = () => {
        if(currentPage.value > 1){
          currentPage.value -= 1;
          emit('change', currentPage.value);
        }
      }
      const nextClick = () => {
        if(currentPage.value < pageCount.value){
          currentPage.value += 1;
          emit('change', currentPage.value);
        }
      }
      const preDisabled = computed(() => {
        return currentPage.value === 1 ? 'disabled' : '';
      })
      const nextDisabled = computed(() => {
        return currentPage.value === pageCount.value ? 'disabled' : '';
      })
      let timer: any = null;
      
      const showSizeChange = (e: Event) => {
        timer = setTimeout(() => {
          if(typeof timer === 'number'){
            clearTimeout(timer)
            const page = (e.target as any)?.value;
            if(typeof page !== 'number' && page >= 1 && page <= pageCount.value) {
              currentPage.value = page;
              emit('change', currentPage.value);
            }
          }
        },300);
      }
        
      return () => {
        return (
          <ul>
            <li class={`pager ${preDisabled.value}`} 
                onClick={preClick}
            >
              <svg focusable="false" class="pre" data-icon="left" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z">
                </path>
              </svg>
            </li>
            {
              pagesList.value.map(pager => pager)
            }
            <li class={`pager ${nextDisabled.value}`}
                onClick={nextClick}>
                <svg focusable="false" class="next" data-icon="right" width="1em" height="1em" fill="currentColor" aria-hidden="true" viewBox="64 64 896 896"><path d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z">
                  </path>
                </svg>
            </li>
            {
              props.showSizeChanger && 
              <div class="show-quick-jumper">
                跳至<input type="text" onInput={showSizeChange} />页
              </div>
            }
          </ul>
        );
      };
    },
});