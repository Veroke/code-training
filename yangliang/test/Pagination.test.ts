import { shallowMount } from "@vue/test-utils";
import { expect, test, vi } from "vitest";
import Pagination from "../src/components/Pagination/Pagination";


test("mount component", () => {
    const wrapper = shallowMount(Pagination, {
      props: {
        current: 1,
        pageSize: 10,
        total: 0,
        showSizeChange: vi.fn(),
        showQuickJumper: false,
        onChange: vi.fn(), 
      },
    });
  
    expect(wrapper.html()).toMatchSnapshot();
});

