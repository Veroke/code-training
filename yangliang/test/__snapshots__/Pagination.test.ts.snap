// Vitest Snapshot v1

exports[`mount component 1`] = `
"<ul>
  <li class=\\"pager disabled\\"><svg focusable=\\"false\\" class=\\"pre\\" data-icon=\\"left\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" viewBox=\\"64 64 896 896\\">
      <path d=\\"M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z\\"></path>
    </svg></li>
  <li class=\\"pager \\"><svg focusable=\\"false\\" class=\\"next\\" data-icon=\\"right\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" viewBox=\\"64 64 896 896\\">
      <path d=\\"M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z\\"></path>
    </svg></li>
  <div class=\\"show-quick-jumper\\">跳至<input type=\\"text\\">页</div>
</ul>"
`;
